src/v/model/tests/random_batch.cc

```diff
@@ -273,7 +274,7 @@ make_random_memory_record_batch_reader(record_batch_spec spec, int n_batches) {
           if (n_batches--) {
               auto batch_spec = spec;
               batch_spec.offset = offset;
-              batches = make_random_batches(batch_spec);
+              batches = make_random_batches(batch_spec).get();
```

Comment:
i think this should not use `get()` as it is not guaranteed that this will be called in the context of seastar thread